pkg/barbican/p11_prep.go

```diff
@@ -40,7 +40,7 @@ func P11PrepJob(instance *barbicanv1beta1.Barbican, labels map[string]string, an
 
 	// add any HSM volumes
 	p11Volumes = append(p11Volumes, GetHSMVolumes(*instance.Spec.PKCS11)...)
-	p11Mounts = append(p11Mounts, GetHSMVolumeMounts(*instance.Spec.PKCS11)...)
+	p11Mounts = append(p11Mounts, GetHSMVolumeMounts()...)
```

Comment:
these volumes/mounts will fail if `*instance.Spec.PKCS11` is nil or the data secret does note exist. see my comment bellow for adding an input validation.

Reply:
Done